googleapiclient/discovery.py

```diff
@@ -950,6 +951,17 @@ def _set_service_methods(self):
     self._add_next_methods(self._resourceDesc, self._schema)
 
   def _add_basic_methods(self, resourceDesc, rootDesc, schema):
+    # If this is the root Resource, add a new_batch_http_request() method.
+    if resourceDesc == rootDesc:
+      def new_batch_http_request(callback=None):
+        batchPath = 'batch'
+        if 'batchPath' in rootDesc:
+          batchPath = rootDesc['batchPath']
```

Comment:
How significant is it that batchPath is computed here instead of outside of the nested method def? Is it expected to change between successive calls to new_batch_http_request?